covsirphy/cleaning/jhu_complement.py

```diff
@@ -199,17 +203,54 @@ def _recovered_full(self, df):
         if df[self.R].max() > self.max_ignored:
             # Necessary if sum of recovered is more than 99%
             # of sum of recovered and infected when outbreaking
-            sel_1 = df[self.C] > self.max_ignored
+            sel_C1 = df[self.C] > self.max_ignored
+            sel_R1 = df[self.R] > self.max_ignored
             sel_2 = df[self.C].diff().diff().rolling(14).mean() > 0
-            s_df = df.loc[
-                sel_1 & sel_2 & (df[self.R] > 0.99 * (df[self.C] - df[self.F]))]
-            if s_df.empty:
+            sel_3 = df[self.R] < 0.01 * (df[self.C] - df[self.F]).rolling(14).mean()
+            s_df_1 = df.loc[
+                sel_C1 & sel_2 & (df[self.R] > 0.99 * (df[self.C] - df[self.F]))]
+            s_df_2 = df.loc[sel_R1 & sel_2 & sel_3]
+            if s_df_1.empty and s_df_2.empty:
                 return df
         # Estimate recovered records
         df[self.R] = (df[self.C] - df[self.F]).shift(
             periods=self.recovery_period, freq="D")
         return df
 
+    def _recovered_partial_ending(self, df):
+        """
+        If ending recovered values do not change for more than applied 'self.interval' days
+        after reached 'self.max_ignored' cases, apply full complement only to these
+        unupdated values and keep the previous valid ones.
+        _recovered_partial() does not handle well such values, because
+        they are not in-between values and interpolation generates only similar values,
+        but small compared to confirmed cases.
+
+        Args:
+            df (pandas.DataFrame):
+                Index: Date (pandas.TimeStamp)
+                Columns: Confirmed, Fatal, Recovered
+
+        Returns:
+            pandas.DataFrame: complemented records
+                Index: Date (pandas.TimeStamp)
+                Columns: Confirmed, Fatal, Recovered
+        """
+        # Whether complement is necessary or not
+        r_max = df[self.R].max()
+        sel_0 = len(np.where(df[self.R] == df[self.R].max())[0]) > self.max_ending_unupdated
```

Comment:
Do you mean `(df[self.R] == df[self.R].max()).sum()` by `len(np.where(df[self.R] == df[self.R].max())[0])`?

Reply:
Yes, applied change!